Wallet API Docs Improvements (#1553)
trinsic-id/sdk: docs/reference/services/wallet-service.md

{{ proto_sample_start() }}
    === "C#"
        ```csharp

>         using Trinsic;

Author: @tmarkovski this is already documented in other parts of the docs. otherwise, we'd have to add the "using Trinsic" and the service instantiation in all the samples. as far as I understand, they're supposed to be snippets specifically of how the methods are called.

            EcosystemId = ecosystem.Id,
            Description = "user123",
            Identity = new CreateWalletRequest.Types.ExternalIdentity {
                Identity = "testemail@trinsic.id",
                Provider = IdentityProvider.Email
            }
        };

        var createWalletResponse = await trinsic.Wallet.CreateWalletAsync(createWalletRequest);

        // Response: {
        //     "authToken": "dGhpcyBpcyBhbiBleGFtcGxlIGF1dGhlbmNpdGlvbiB0b2tlbgo=",
        //     "tokenId": "0b4f42cb-4d44-4629-89dd-47b814229ffe",
        //     "wallet": {
        //          "walletId": "urn:trinsic:wallets:z7438uW5X4gZ1rZsiZaBdxX",
        //          "publicDid": "did:key:123456"
        //     }

>         // }

Reviewer: I would argue that this can be omitted since we already make the response model available in one of the other tabs. Unless you feel strongly about this, I'd remove it.

Reviewer: This comment apply to other places that also include a similar comment in the snippet

Author: I agree with you :) Will remove them

---

## Authenticate

Authenticate and return an auth token for an    existing wallet using one of the associated external identities.

> This endpoint requires that the wallet user has previously added at least one external identity using the above endpoints.

Reviewer 2: ```suggestion
This endpoint requires that the wallet user has previously added at least one external identity using `AddExternalIdentity` calls below.
```

Or mention below, since they are below in this document. But probably better to not mention above/below as it'd be lost in a restructure of the docs.

Author: thanks! changed it to "calls" :)

Once a token is obtained, it can be reused for future sessions -- users don't need to authenticate if they already have a valid token.
You can store the auth token in secure enclaves on the users device, browser, etc.

!!! question "When should users authenticate?"

    - If your integration solution doesn't manage the wallet tokens, users may need to re-authenticate on their device to get a new auth token

>     - Users want to log in to a different device using their email or phone number

Reviewer 2: ```suggestion
    - Users that want to log in to a different device using their email or phone number
```

>     - Returning users that have lost their previous session and require new auth token

Reviewer 2: ```suggestion
    - Returning users that have lost their previous session and require a new auth token
```

### AuthenticateInit

{{ proto_sample_start() }}
    === "C#"
        ```csharp

>         // Step 1 - initiate authentication challenge

Reviewer 3: Same comment here on removing using statements.

Author: Answered in the other comment hahah

        var authenticateInitRequest = new AuthenticateInitRequest {
            Identity = "test@trinsic.id",
            Provider = IdentityProvider.Email,
            EcosystemId = "test-ecosystem" // short name or full ecosystem ID
        };
        var authenticateInitResponse = await trinsic.Wallet.AuthenticateInitAsync(authenticateInitRequest);
        ```

{{ proto_method_tabs("services.universalwallet.v1.UniversalWallet.AuthenticateInit") }}

### AuthenticateConfirm

{{ proto_sample_start() }}
    === "C#"
        ```csharp
        // Step 2 - confirm authentication response
        var authenticateConfirmRequest = new AuthenticateConfirmRequest {
            Challenge = authenticateInitResponse.Challenge,
            Response = "123456" // OTP code
        };
        var authenticateConfirmResponse = await trinsic.Wallet.AuthenticateConfirmAsync(authenticateConfirmRequest);

        // Response: {
        //     "authToken": "dGhpcyBpcyBhbiBleGFtcGxlIGF1dGhlbmNpdGlvbiB0b2tlbgo="
        // }

        // use the new token to make authenticated calls

>         // var options = new TrinsicOptions { AuthToken = AuthenticateConfirmResponse.AuthToken };

Reviewer 2: ```suggestion
        // var options = new TrinsicOptions { AuthToken = authenticateConfirmResponse.AuthToken };
```

        // trinsic = new TrinsicService(options);
        ```

{{ proto_method_tabs("services.universalwallet.v1.UniversalWallet.AuthenticateConfirm") }}

---

## Add External Identity

> This service is used to attach external identity, such as email or phone number, to a wallet. The purpose of this process is to allow

Reviewer 2: ```suggestion
This service is used to attach an external identity, such as an email or a phone number, to a wallet. The purpose of this process is to allow
```
